pkg/nsx/services/vpc/vpc.go

```diff
@@ -471,3 +523,236 @@ func (s *VPCService) Cleanup() error {
 
 	return nil
 }
+
+func (service *VPCService) needUpdateRule(rule *model.Rule, externalCIDR []string) bool {
+	des := rule.DestinationGroups
+	currentDesSet := sets.Set[string]{}
+	for _, group := range des {
+		currentDesSet.Insert(group)
+	}
+	if len(externalCIDR) != len(currentDesSet) {
+		return true
+	}
+	for _, newGroup := range externalCIDR {
+		if !currentDesSet.Has(newGroup) {
+			return true
+		}
+	}
+	return false
+}
+
+func (service *VPCService) getExternalIpBlock(vpcName string, namespace string) ([]string, error) {
+	// read corresponding vpc network config from store
+	nc_name, err := service.getNetworkconfigNameFromNS(namespace)
```

Comment:
nc_name is a sneak naming convention, it's good that we keep camel naming convention in our code.

Reply:
done